### PR TITLE
update Iron Tanks

### PR DESCRIPTION
**Pull request by seitenca**
Updated tank textures
Added Emerald Tank (96 buckets)
https://www.curseforge.com/minecraft/mc-mods/iron-tanks/files/2708845

### CODE REVIEW

File: src/main/java/com/indemnity83/irontank/IronTank.java [hunk 1]
```diff
@@ -27,8 +24,6 @@
 	@SidedProxy(clientSide = Reference.CLIENT_PROXY_CLASS, serverSide = Reference.SERVER_PROXY_CLASS)
 	public static IProxy proxy;
 
-	public static ItemMapHelper itemMapHelper;
```
Comment (bartimaeusnek) [inline src/main/java/com/indemnity83/irontank/IronTank.java:30]: Why was this removed?

File: src/main/java/com/indemnity83/irontank/reference/TankChangerType.java [hunk 2]
```diff
@@ -59,10 +51,10 @@
 	 * @return true if upgrade is allowed
 	 */
 	public boolean canUpgrade(TankType from) {
```
Comment (bartimaeusnek) [inline src/main/java/com/indemnity83/irontank/reference/TankChangerType.java:53]: doesnt this allows ALL upgrades to be applied to ANY type?

File: src/main/java/com/indemnity83/irontank/reference/TankType.java [hunk 3]
```diff
@@ -24,14 +16,14 @@
...
-    OBSIDIAN(64, "obsidiantank", Arrays.asList("obsidian"), Arrays.asList("tgtg3gtgt"), 6000000.0F),
-    GLASS(0, "", Arrays.asList("blockGlass"), Arrays.asList(""), 0f),
-	EMERALD(96, "emeraldtank", Arrays.asList("gemEmerald"), Arrays.asList("gggt3tggg"), 25/3F);
+    IRON(32, "ironTank", Arrays.asList("ingotIron", "ingotRefinedIron"), Arrays.asList("tgtg0gtgt", "gggt4tggg"), 25/3F),
+    GOLD(48, "goldTank", Collections.singletonList("ingotGold"), Arrays.asList("tgtg1gtgt", "gggt5tggg"), 25/3F),
+    DIAMOND(64, "diamondTank", Collections.singletonList("gemDiamond"), Arrays.asList("gggt2tggg", "gggg5gttt"), 25/3F),
+    COPPER(27, "copperTank", Collections.singletonList("ingotCopper"), Collections.singletonList("tgtg0gtgt"), 25/3F),
+    SILVER(43, "silverTank", Collections.singletonList("ingotSilver"), Arrays.asList("tgtg4gtgt", "gggt1tggg"), 25/3F),
+    OBSIDIAN(64, "obsidianTank", Collections.singletonList("obsidian"), Collections.singletonList("tgtg3gtgt"), 6000000.0F),
+	EMERALD(96, "emeraldTank", Collections.singletonList("gemEmerald"), Collections.singletonList("gggt3tggg"), 25/3F),
```
Comment (bartimaeusnek) [inline src/main/java/com/indemnity83/irontank/reference/TankType.java]: 25F/3F please, for all of them.

File: src/main/java/com/indemnity83/irontank/utility/ItemMapHelper.java [hunk 4]
```diff
@@ -1,49 +0,0 @@
...
-
-        remapList.put("irontank:copperTank", Item.getItemFromBlock(ModBlocks.copperTank));
-        remapList.put("irontank:ironTank", Item.getItemFromBlock(ModBlocks.ironTank));
-        remapList.put("irontank:silverTank", Item.getItemFromBlock(ModBlocks.silverTank));
-        remapList.put("irontank:goldTank", Item.getItemFromBlock(ModBlocks.goldTank));
-        remapList.put("irontank:obsidianTank", Item.getItemFromBlock(ModBlocks.obsidianTank));
-        remapList.put("irontank:diamondTank", Item.getItemFromBlock(ModBlocks.diamondTank));
-    }
-
-    public void handleMissingMaps(FMLMissingMappingsEvent event) {
```
Comment (bartimaeusnek) [inline src/main/java/com/indemnity83/irontank/utility/ItemMapHelper.java:34]: I still dont get why you removed this.

**Review by bartimaeusnek — CHANGES_REQUESTED** [review]
please resolve the comments i wrote, also i dont really get the point of changing the names...

File: src/main/java/com/indemnity83/irontank/IronTank.java (hunk 1 above)
Comment (seitenca) [reply src/main/java/com/indemnity83/irontank/IronTank.java:30]: This is removed because I have reberted all the items id changes

Comment (seitenca) [reply src/main/java/com/indemnity83/irontank/IronTank.java:30]: So now its the same id as before

File: src/main/java/com/indemnity83/irontank/reference/TankType.java (hunk 3 above)
Comment (seitenca) [reply src/main/java/com/indemnity83/irontank/reference/TankType.java]: Ok, but why?

File: src/main/java/com/indemnity83/irontank/reference/TankChangerType.java (hunk 2 above)
Comment (seitenca) [reply src/main/java/com/indemnity83/irontank/reference/TankChangerType.java]: ?

File: src/main/java/com/indemnity83/irontank/utility/ItemMapHelper.java (hunk 4 above)
Comment (seitenca) [reply src/main/java/com/indemnity83/irontank/utility/ItemMapHelper.java:34]: This class is for not loss the items after the items id changes

Comment (seitenca) [reply src/main/java/com/indemnity83/irontank/utility/ItemMapHelper.java:34]: Now it is useless because I have reberted the items id back

File: src/main/java/com/indemnity83/irontank/reference/TankChangerType.java (hunk 2 above)
Comment (seitenca) [reply src/main/java/com/indemnity83/irontank/reference/TankChangerType.java]: What change you say?

File: src/main/java/com/indemnity83/irontank/reference/TankType.java (hunk 3 above)
Comment (Dream-Master) [reply src/main/java/com/indemnity83/irontank/reference/TankType.java]: Obsidian is Blast proof.

Comment (seitenca) [reply src/main/java/com/indemnity83/irontank/reference/TankType.java]: Yes thats why i ask

File: src/main/java/com/indemnity83/irontank/reference/TankChangerType.java (hunk 2 above)
Comment (seitenca) [reply src/main/java/com/indemnity83/irontank/reference/TankChangerType.java:53]: this in an idea inspection that reverted this also
https://github.com/seitenca/irontanks/blob/support/1.7.10/src/main/java/com/indemnity83/irontank/item/ItemTankChanger.java
!type.canUpgrade -> type.canUpgrade

File: src/main/java/com/indemnity83/irontank/reference/TankType.java (hunk 3 above)
Comment (bartimaeusnek) [reply src/main/java/com/indemnity83/irontank/reference/TankType.java]: I meant, you have `25/3F`  for most cases, thats an integer that gets auto converted into a float, please make that to an explicit float division. (`25F/3F`)

File: src/main/java/com/indemnity83/irontank/reference/TankChangerType.java (hunk 2 above)
Comment (bartimaeusnek) [reply src/main/java/com/indemnity83/irontank/reference/TankChangerType.java]: `==` into `!=`

File: src/main/java/com/indemnity83/irontank/reference/TankType.java (hunk 3 above)
Comment (seitenca) [reply src/main/java/com/indemnity83/irontank/reference/TankType.java]: one question bart you know why in gtnh dont work the blast proof resistance of the obsidian tank?

Comment (bartimaeusnek) [reply src/main/java/com/indemnity83/irontank/reference/TankType.java]: tectech is nerfing that.
